### DISCUSSION TITLE
Selecting a "main" drainage divide using DIVIDEobj

### DISCUSSION THREAD

**Discussion by caiocrelier** [wschwanghart/topotoolbox#34] [Questions] [answered]
Hello, Wolfgang! I have been using your tools for a couple of years now, and I must say, I really appreciate the great work you've been doing.

I have a selection of DEM images comprising a big portion of the Brazilian landscape (about 150000 km² each). These DEMs contain smaller parts of a bigger water basin divide. I am wondering if there is a better or easier way to select and analyze the drainage divide asymmetry and the direction of the hillslope of only those divides, as shown in the image. It could be by "cleaning" the rest of the DEM after the calculation is done, or by selecting the divide and running the code only in the selected areas. Do you have any insight or have you ever done that in such a big area, where selecting basins two by two would be quite time-consuming?

![exemplo](https://user-images.githubusercontent.com/49659628/235796548-a096e8e4-5e7d-40a6-b780-264757aa5b1d.png)

Thank you!

**Comment by dirkscherler** [comment, marked as answer]
Hi Caio,
I'm answering here, as you ask about the DIVIDEobj. The divides that are calculated are defined by a stream network, or individual streams, which can be provided as either a STREAMobj or a boolean GRIDobj. Check out the help of the DIVIDEobj function. That means you can control which divides will be calculated by providing a specific stream network. There exist several functions to modify stream networks, including "modify" (for STREAMobj's). That's probably the way I would approach this problem. I would test around with the Big Tujunga example and once I found a way that seems reasonable, I would give it a try on your big DEM.
Hope that helps.
Cheers,
Dirk

**Reply by caiocrelier** [reply]
Thank you for your reply, Dirk! I will check the modify function.
